x-pack/plugins/endpoint/server/routes/whitelist.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { schema } from '@kbn/config-schema';
+import { SearchResponse } from 'elasticsearch';
+import { EndpointAppContext } from '../types';
+import { WhitelistRule } from '../../common/types';
+
+const whitelistIdx = 'whitelist-index';  // TODO: change this
+
+/**
+ * Registers the whitelist routes for the API
+ */
+export function registerWhitelistRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  router.get(
+    {
+      path: '/api/endpoint/whitelist',
+      validate: {},
+      options: { authRequired: true },
+    },
+    handleWhitelistGet
+  );
+
+  const INVALID_UUID = "`alert_id` Must be a UUID.'";
+  router.post(
+    {
+      path: '/api/endpoint/whitelist',
+      validate: {
+        body: schema.object({
+          comment: schema.maybe(schema.string()), // Optional comment explaining reason for whitelist
+          alert_id: schema.string({
+            minLength: 36, // https://tools.ietf.org/html/rfc4122#section-3
+            maxLength: 36,
+            validate(value) {
+              // Must be a UUID
+              if (!validateUUID(value)) {
+                return INVALID_UUID;
```

Comment:
Let's use i18n for error messages like here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/endpoint/server/routes/alerts/list/schemas.ts#L42

Reply:
thanks